server/src/main/java/org/elasticsearch/common/time/DateFormatter.java

```diff
@@ -87,7 +86,8 @@ default DateTime parseJoda(String input) {
      * Return the given millis-since-epoch formatted with this format.
      */
     default String formatMillis(long millis) {
-        return format(ZonedDateTime.ofInstant(Instant.ofEpochMilli(millis), ZoneOffset.UTC));
+        ZoneId zone = zone() != null ? zone() : ZoneOffset.UTC;
```

Comment:
I see the previous behaviour was to use UTC. But maybe we should default to ZoneId.systemDefault ?

Reply:
why should we do that? This would mean we end up with different dates on differently configured systems (which the person running ES on that system might not be able to control)? UTC here guarantees us to have consistent dates and  requires the user to configure the time zone explicitely

Reply:
that makes sense, thanks for clarification